liteshort.py

```diff
@@ -117,18 +117,15 @@ def check_short_exist(short):  # Allow to also check against a long link
     return False
 
 
-def check_self_link(long):
-    if get_baseUrl().rstrip("/") in long:
-        return True
-    return False
-
-
 def linking_to_blacklist(long):
     # Removes protocol and other parts of the URL to extract the domain name
     long = long.split("//")[-1].split("/")[0]
-    if long in current_app.config["blacklist"]:
-        return True
-    return False
+    if not current_app.config["selflinks"]:
+        return (
+            long in get_baseUrl()
+            or long in current_app.config["blacklist"]
+        )
+    return long in current_app.config["blacklist"]
```

Comment:
This is nitpicky at this point (especially considering the code quality of the rest of the project), but I think this should be simplified to this:
```python3
if long in current_app.config["blacklist"]: # first, since this condition is evaluated whether selflinks or not
    return False
if not current_app.config["selflinks"]:
    return long in get_baseUrl()
```

Other than that, looks good.